File: backend/model/src/main/kotlin/classes/VideoGame.kt
package classes

import java.sql.Date
import javax.persistence.*

@Entity
@Table(name = "video_games")
class VideoGame(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int,

    @Column(name = "title", nullable = false)
    val title: String,

    @Column(name = "release_date", nullable = true)
    val releaseDate: Date?,

    @ManyToMany
    @JoinTable(
        name = "video_game_genre",
        joinColumns = [JoinColumn(name = "video_game_id")],
        inverseJoinColumns = [JoinColumn(name = "genre_id")]
    )
    val genres: Set<Genre>?
)

File: frontend/src/app/model/VideoGame.ts
import {Genre} from "./Genre";
import {Platform} from "./Platform";
import {Developer} from "./Developer";
import {VideoGameUser} from "./VideoGameUser";

export interface VideoGame {
  id: number
  title: string
  releaseDate: Date
  description: string
  mainImage: string
  genres?: Genre[]
  platforms?: Platform[]
  developer?: Developer
  videoGameUsers: VideoGameUser[]
}
